src/pages/spawn/spawnHost/SpawnHostModal.tsx

```diff
@@ -177,7 +177,7 @@ export const SpawnHostModal: React.FC<SpawnHostModalProps> = ({
   const canSubmitSpawnHost = !(
     distroId === "" ||
     region === "" ||
-    publicKey.key === ""
+    publicKey?.key === ""
```

Comment:
does this make more sense `!publicKey?.key` in case `publicKey?.key` evaluates to undefined?

Reply:
key is required so it should always be defined 
https://github.com/evergreen-ci/evergreen/blob/d74f1e7c8d3b5d19538fd5eefde753680cf672f7/graphql/schema.graphql#L825